system_tests/monitoring.py

```diff
@@ -175,3 +177,145 @@ def test_create_and_delete_metric_descriptor(self):
         descriptor.delete()
         with self.assertRaises(NotFound):
             descriptor.delete()
+
+
+class TestMonitoringGroups(unittest2.TestCase):
+
+    def setUp(self):
+        self.to_delete = []
+        self.DISPLAY_NAME = 'Testing: New group'
+        self.FILTER = 'resource.type = "gce_instance"'
+        self.IS_CLUSTER = True
+
+    def tearDown(self):
+        for group in self.to_delete:
+            backoff_intervals = [1, 2, 4, 8]
+            while True:
+                try:
+                    group.delete()
+                    break
+                except NotFound:
+                    if backoff_intervals:
+                        time.sleep(backoff_intervals.pop(0))
+                    else:
+                        raise
```

Comment:
Yes - this is used in the system test for both logging and bigtable.

Retrying deletion is important, otherwise we might pollute our test project even if all the tests pass fine.